src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -433,43 +430,50 @@ bool InputStateMachineEngine::ActionCsiDispatch(const VTID id, const VTParameter
     case CsiActionCodes::CSI_F1:
     case CsiActionCodes::CSI_F2:
     case CsiActionCodes::CSI_F4:
-        success = _GetCursorKeysVkey(id, vkey);
-        modifierState = _GetCursorKeysModifierState(parameters, id);
-        success = success && _WriteSingleKey(vkey, modifierState);
-        break;
+    {
+        short vkey = 0;
+        if (_GetCursorKeysVkey(id, vkey))
+        {
+            const auto modifierState = _GetCursorKeysModifierState(parameters, id);
+            _WriteSingleKey(vkey, modifierState);
+        }
+        return true;
+    }
     case CsiActionCodes::Generic:
-        success = _GetGenericVkey(parameters.at(0), vkey);
-        modifierState = _GetGenericKeysModifierState(parameters);
-        success = success && _WriteSingleKey(vkey, modifierState);
-        break;
+    {
+        short vkey = 0;
+        if (_GetGenericVkey(parameters.at(0), vkey))
+        {
+            const auto modifierState = _GetGenericKeysModifierState(parameters);
+            _WriteSingleKey(vkey, modifierState);
+        }
+        return true;
+    }
     case CsiActionCodes::CursorBackTab:
-        success = _WriteSingleKey(VK_TAB, SHIFT_PRESSED);
-        break;
+        _WriteSingleKey(VK_TAB, SHIFT_PRESSED);
+        return true;
     case CsiActionCodes::DTTERM_WindowManipulation:
-        success = _pDispatch->WindowManipulation(parameters.at(0), parameters.at(1), parameters.at(2));
-        break;
+        _pDispatch->WindowManipulation(parameters.at(0), parameters.at(1), parameters.at(2));
+        return true;

```

Comment:
Do we actually still need this? AFAICT, conpty now reports changes in the terminal's window size through the `PtySignalInputThread`, but I'm not sure whether any of the other operations here still serve a purpose. And if we can get rid of this, then the window size queries like `^[[18t` should start working again in the cmd shell.